### DISCUSSION TITLE
Weak password report freezes browser

### DISCUSSION THREAD

**Discussion by johanvdw** [dani-garcia/vaultwarden#3576] [Q&A]
We are using vaultwarden 1.28.1 using the provided docker image. When we open the weak password report, the browser freezes completely, without giving any sensible feedback. I tried both Chrome and firefox.

Websockets are enabled and our proxied through nginx.

Is this working correctly for other people?

**Comment by johanvdw** [comment]
I can not reproduce it on vault.vaultwarden.net

**Comment by BlackDex** [comment]
What does the developer console of the browser report? (F12).

**Comment by johanvdw** [comment]
Nothing is reported in the console. These messages show before opening the page though:
![image](https://github.com/dani-garcia/vaultwarden/assets/868434/fd552a9a-eabd-451d-a1ec-5c730c37045c)
The token 400 appears before asking for the TOTP code, so I think it is unrelated. Other things work afterwards.

**Comment by BlackDex** [comment]
I just checked my instances, and that also takes a while. That probably is because it needs to check a lots of passwords.
Depending on the total amount of passwords you have it can take a while.

This is not something we can fix on the server side at Vaultwarden, since this is all client side handled.

**Reply by johanvdw** [reply]
I can confirm that on the same instance, it works with a user with only access to a limited number of passwords. It also does not seem to scale linearly with the number of passwords.

**Reply by BlackDex** [reply]
I suggest to create an issue at Bitwarden regarding this.

**Reply by johanvdw** [reply]
did that https://github.com/bitwarden/clients/issues/5603

**Comment by Infinoid** [comment]
I noticed this today with the latest vaultwarden 1.35.4 release.  Firefox gave me this warning:

<img width="769" height="45" alt="image" src="https://github.com/user-attachments/assets/b75f28bb-df32-4018-bb67-e8de1ea638da" />

After 5 minutes, I gave up and stopped it.  Firefox became responsive again after closing the tab.
